pkg/disk/disk.go

```diff
@@ -0,0 +1,138 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+// Package disk manages the persistent disk used to save containerd user data
+package disk
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/fs"
+
+	"github.com/spf13/afero"
+
+	"github.com/runfinch/finch/pkg/command"
+	"github.com/runfinch/finch/pkg/path"
+)
+
+const (
+	// diskName must always be consistent with the value under additionalDisks in finch.yaml.
+	diskName = "finch"
+)
+
+// UserDataDiskManager is used to check the user data disk configuration and create it if needed.
+type UserDataDiskManager interface {
+	InitializeUserDataDisk() error
+}
+
+// fs functions required for setting up the user data disk.
+type diskFS interface {
+	afero.Fs
+	afero.Linker
+	afero.LinkReader
+}
+
+type userDataDiskManager struct {
+	lcc     command.LimaCmdCreator
+	fs      diskFS
+	finch   path.Finch
+	homeDir string
+}
+
+// NewUserDataDiskManager is a constructor for UserDataDiskManager.
+func NewUserDataDiskManager(
+	lcc command.LimaCmdCreator,
+	fs diskFS,
+	finch path.Finch,
+	homeDir string,
+) UserDataDiskManager {
+	return &userDataDiskManager{
+		lcc:     lcc,
+		fs:      fs,
+		finch:   finch,
+		homeDir: homeDir,
+	}
+}
+
+// InitializeUserDataDisk checks the current disk configuration and fixes it if needed.
+func (m *userDataDiskManager) InitializeUserDataDisk() error {
```

Comment:
nit: might want to change this name to something that is more descriptive of the behavior. Without reading the code, I would expect that this just does first time setup rather than also fixing existing disks/configurations. If you agree, maybe something like `EnsureUserDataDisk` would be more appropriate?